app/src/pages/prompt/PromptChatMessagesCard.tsx

```diff
@@ -87,15 +105,21 @@ function ChatMessages({
   const { messages } = template;
   return (
     <Flex direction="column" gap="size-200">
-      {messages.map((message, i) => (
-        // TODO: Handle JSON content for things like tool calls
-        <ChatTemplateMessage
-          key={i}
-          role={message.role as string}
-          content={message.content || JSON.stringify(message.jsonContent)}
-          templateFormat={templateFormat}
-        />
-      ))}
+      {messages.map((message, i) => {
+        const textPart = message.content.find(isTextPart);
+        // TODO(apowell): Break out into switch statement, rendering each message part type
```

Comment:
It might be worth filing an issue for this and putting it in the TODO so you can search for it.